rmf_fleet_adapter/src/rmf_fleet_adapter/agv/RobotUpdateHandle.cpp

```diff
@@ -272,5 +286,38 @@ void RobotUpdateHandle::Unstable::set_lift_entry_watchdog(
   }
 }
 
+//==============================================================================
+void RobotUpdateHandle::ActionExecution::update_remaining_time(
+    rmf_traffic::Duration remaining_time_estimate)
+{
+  _pimpl->data->remaining_time = remaining_time_estimate;
+}
+
+//==============================================================================
+void RobotUpdateHandle::ActionExecution::finished()
+{
+  _pimpl->data->finished();
```

Comment:
Oh, and one other final detail. We should make sure that the finished callback can only get triggered exactly once. We can do that like so:

```
if (!_pimpl->data->finished)
  return;

_pimpl->data->finished();
_pimpl->data->finished = nullptr;
```

Reply:
Done 👍🏼 
https://github.com/open-rmf/rmf_ros2/pull/155/commits/22a4fbae3018a9f3eaafed614731857c822d5a25